editor/project_manager.cpp

```diff
@@ -904,36 +904,53 @@ class ProjectDialog : public ConfirmationDialog {
 			default_renderer_type = EditorSettings::get_singleton()->get_setting("project_manager/default_renderer");
 		}
 
+#if defined(GLES3_ENABLED) && (defined(WEB_ENABLED) || !defined(VULKAN_ENABLED))
+		// Web platform only supports `gl_compatibility`. Ignore the above editor setting's value.
+		default_renderer_type = "gl_compatibility";
+#endif
+
 		Button *rs_button = memnew(CheckBox);
 		rs_button->set_button_group(renderer_button_group);
 		rs_button->set_text(TTR("Forward+"));
+#if !defined(VULKAN_ENABLED) || defined(WEB_ENABLED)
+		rs_button->set_disabled(true);
+#endif
 		rs_button->set_meta(SNAME("rendering_method"), "forward_plus");
 		rs_button->connect("pressed", callable_mp(this, &ProjectDialog::_renderer_selected));
 		rvb->add_child(rs_button);
+#if defined(VULKAN_ENABLED)
 		if (default_renderer_type == "forward_plus") {
 			rs_button->set_pressed(true);
 		}
-
+#endif
 		rs_button = memnew(CheckBox);
 		rs_button->set_button_group(renderer_button_group);
 		rs_button->set_text(TTR("Mobile"));
+#if !defined(VULKAN_ENABLED) || defined(WEB_ENABLED)
+		rs_button->set_disabled(true);
+#endif
 		rs_button->set_meta(SNAME("rendering_method"), "mobile");
 		rs_button->connect("pressed", callable_mp(this, &ProjectDialog::_renderer_selected));
 		rvb->add_child(rs_button);
+#if defined(VULKAN_ENABLED)
```

Comment:
I think this shouldn't be needed, it's the `default_renderer_type` selection which should be fixed.

But I think it's overkill and won't play nice with Direct3D12, Metal or WebGPU so I'd keep it simpler.

Reply:
Done. I've kept the `GLES3_ENABLED` checks for disabling the Compatibility option, since it's unlikely it will be implemented using another rendering driver.